### PR TITLE
Shift+F3 debug menu

### PR DESCRIPTION
**Pull request by AffectedArc07**
## What Does This PR Do
Adds a debug menu that toggles when you press Shift+F3. I wish I could make it regular `F3` for reasons, but for some inane reason that key is bound to say.

This menu shows the stats for SSs which are likely to lag the game. We dont need the full list, and this is enough. It also doesn't lag the client (or the server) at all, whereas the existing MC tab straight up hardcrashes my client on prod most of the time, which is kind of an issue when you're the host

## Why It's Good For The Game
See above. The MC tab is bad, but we aren't moving to TG statpanes, because they're flawed in several ways.

## Images of changes
image

## Changelog
:cl: AffectedArc07
add: [Admin] You can now view a debug menu with SHIFT+F3
/:cl:

*miningggg awayyyyyyyyyyyyyyy*

### CODE REVIEW

File: code/controllers/subsystem/debugview.dm [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	var/mty = 480 - 9 * length(entries)
+
+	// And update the clients
+	for(var/client/C as anything in processing)
+		C.debug_text_overlay.maptext_y = mty
+		C.debug_text_overlay.maptext = MAPTEXT(out_text)
+
+/datum/controller/subsystem/debugview/proc/start_processing(client/C)
+	C.debug_text_overlay = new /obj/screen/fullscreen
+	// DO NOT CHANGE THE BELOW VALUES
```
Comment (moxian) [inline code/controllers/subsystem/debugview.dm:43]: This is not a very helpful comment for future readers. Consider something like:
```suggestion
	// Give maptext the full screen to print over,
	// and layer it on top of literally everything else.
```

File: code/controllers/subsystem/debugview.dm [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	var/out_text = entries.Join("\n")
+	var/mty = 480 - 9 * length(entries)
+
+	// And update the clients
+	for(var/client/C as anything in processing)
+		C.debug_text_overlay.maptext_y = mty
+		C.debug_text_overlay.maptext = MAPTEXT(out_text)
+
+/datum/controller/subsystem/debugview/proc/start_processing(client/C)
+	C.debug_text_overlay = new /obj/screen/fullscreen
```
Comment (moxian) [inline code/controllers/subsystem/debugview.dm:42]: using `/obj/screen/fullscreen` here means it's treated the same way as blindness overlay, or the red low-hp overlay; in particular they get cleared when the client gets transfered to a new mob (relevant to admins aghosting).
Should this maybe be `/obj/screen` instead, or even a dedicated `/obj/screen/debugview` subtype?

**Comment by Sirryan2002** [comment]
I personally do not have issues loading the MC, but I know others have serious issues with it. This is a great usability feature for people 👍

File: code/controllers/subsystem/debugview.dm (hunk 2 above)
Comment (AffectedArc07) [reply code/controllers/subsystem/debugview.dm:42]: I had noticed that, but I wanted to inherit a screen with the same size on everything. Will make it into a proper type.

Comment (AffectedArc07) [reply code/controllers/subsystem/debugview.dm:42]: Making it into its own type still has the issues of mob switching removing it, despite me binding it to client screens.

**Comment by AffectedArc07** [comment]
image

Added a backdrop

**Comment by AffectedArc07** [comment]
This PR escalated a bit, and partly refactored SS info reporting.

I was testing displaying al SSs, but that was deemed to be awful.
image

I could make an `SS_NO_DEBUGVIEW` flag, but that doesnt fix issues with air overhanging
image


And since I already did the refactoring, I made an `SS Info Breakdown` verb.
image

I can still view the entire SS info without looking at the MC tab, and this I no longer have to crash my game.

File: code/controllers/subsystem.dm [hunk 3]
```diff
@@ -172,20 +175,24 @@
...
 		title = "[state_colour()]\[[state_letter()]][title]</font>"
+	else
+		title = "\[O][title]"
 
 	stat(title, statclick.update(msg))
 
 /datum/controller/subsystem/proc/state_letter()
+	if(!can_fire)
+		return "O"
 	switch(state)
```
Comment (AffectedArc07) [inline code/controllers/subsystem.dm:196]:
```suggestion
/datum/controller/subsystem/proc/state_letter()
	switch(state)
```

Redundant change

File: code/controllers/subsystem/debugview.dm [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+/datum/controller/subsystem/debugview/proc/start_processing(client/C)
+	C.debug_text_overlay = new /obj/screen/debugtextholder
+	C.screen |= C.debug_text_overlay
+	processing |= C
+
+/datum/controller/subsystem/debugview/proc/stop_processing(client/C)
+	C.screen -= C.debug_text_overlay
+	qdel(C.debug_text_overlay)
+	processing -= C
```
Comment (AffectedArc07) [inline code/controllers/subsystem/debugview.dm:49]:
```suggestion
/datum/controller/subsystem/debugview/proc/stop_processing(client/C)
	processing -= C
	C.screen -= C.debug_text_overlay
	qdel(C.debug_text_overlay)
```
Heaven forbid if BYOND ever goes async

File: code/__DEFINES/layers.dm [hunk 5]
```diff
@@ -129,5 +129,8 @@
 // This should always be on top.
 #define HUD_PLANE_BUILDMODE 30
 
+// Nah, this should always be top
+#define HUD_PLANE_DEBUGVIEW 40
+
```
[the remark this answers is not in the document]
Comment (AffectedArc07) [reply code/__DEFINES/layers.dm:134]: Yes

File: code/controllers/subsystem/garbage.dm [hunk 6]
```diff
@@ -41,25 +41,26 @@ SUBSYSTEM_DEF(garbage)
...
-/datum/controller/subsystem/garbage/stat_entry(msg)
+/datum/controller/subsystem/garbage/get_stat_details()
+	var/list/msg = list()
 	var/list/counts = list()
 	for(var/list/L in queues)
 		counts += length(L)
-	msg += "Queue:[counts.Join(",")] | Del's:[delslasttick] | Soft:[gcedlasttick] |"
-	msg += "GR:"
+	msg += "Q:[counts.Join(",")] | D:[delslasttick] | S:[gcedlasttick] |"
+	msg += "GCR:"
```
[the remark this answers is not in the document]
Comment (AffectedArc07) [reply code/controllers/subsystem/garbage.dm:50]: Ill space them

**Review by hal9000PR — APPROVED** [review]
Lgtm

**Comment by AffectedArc07** [comment]
Removed codeowners review since it changes no logic

**Comment by hal9000PR** [comment]
monolith crashing the MC

**Pull request merged**